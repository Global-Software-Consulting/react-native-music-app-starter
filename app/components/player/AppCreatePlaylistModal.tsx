import React from 'react';
import { View, TouchableOpacity, Button, TextInput } from 'react-native';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import { useDispatch, useSelector } from 'react-redux';
import { useTheme, Text } from 'react-native-paper';
import useStyles from './styles';
import Modal from "react-native-modal";
import AppHeader from '../AppHeader';
import { updatePlayList } from '../../store/actions/playerActions';
import { IPlayerState } from '../../models/reducers/player';

interface MusicProps {
  isCreateModalVisible?: boolean;
  closeModals?: any;

}
interface IState {
  playerReducer: IPlayerState;

}
const AppCreatePlaylistModal: React.FC<MusicProps> = ({
  isCreateModalVisible,
  closeModals
}) => {
  
  const styles = useStyles();
  const theme = useTheme();
  const [name, onChangeName] = React.useState('');
  const item = useSelector((state: IState) => state.playerReducer.playerList);
  const playlist = useSelector((state: IState) => state.playerReducer.playList);
  const dispatch = useDispatch();

  const saveNewPlayList = () => {

    let data = playlist;
    let found = playlist?.find((element: any) => element.name == name);
    if (!found) {
      let obj = { name: name, songs: [item] };
      data.push(obj);
      dispatch(updatePlayList(data));
      closeModals();
    } else {
      console.log("name already exists.....");
    }

  }
  return (
    
      <View style={styles.container}>
        <Modal isVisible={isCreateModalVisible}>
          <View style={styles.CreateModalContainer}>
            <AppHeader
              renderLeft={
                <View style={{ flexDirection: 'row', padding: 8 }}>
                  <TouchableOpacity onPress={() => closeModals()}>
                    <MaterialCommunityIcons
                      name="close"
                      size={30}
                      color={theme.colors.primary}
                    />
                  </TouchableOpacity>

                  <Text style={styles.label}>Create Playlist</Text>
                  <View style={{ width: '10%' }} />
                  <TouchableOpacity onPress={saveNewPlayList} style={styles.saveButton}>

                    <Text style={styles.saveButtonText}>Save</Text>
                  </TouchableOpacity>

                </View>
              } />
            <View style={styles.inputWrapper}>
              <Text>Playlist:</Text>
              <TextInput
                placeholder="Type here.."
                style={styles.input}
                value={name}
                onChangeText={(name) => onChangeName(name)}
                maxLength={10}
                keyboardType="default"
              />
            </View>
          </View>
        </Modal>
      </View>
    
  );
};

export default AppCreatePlaylistModal;
