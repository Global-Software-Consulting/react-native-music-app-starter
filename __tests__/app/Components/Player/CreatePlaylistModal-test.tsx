/**
 * @format
 */
import 'react-native';
import React from 'react';
import AppCreatePlaylistModal from '../../../../app/components/player/AppCreatePlaylistModal';
// Note: test renderer must be required after react-native.
import { render, fireEvent } from '@testing-library/react-native';


jest.mock("react-native-responsive-screen", () => {
    return {
        heightPercentageToDP: jest.fn(),
        widthPercentageToDP: jest.fn(),

    }
});

jest.mock('@react-navigation/native', () => {
    return {
        useIsFocused: jest.fn(),
        useNavigation: jest.fn(),
        // useRoute: jest.fn(),
        useRoute: () => {
            return {
                route: {
                    params: {
                        item: {
                            id: '5',
                            url: 'https://www.chosic.com/wp-content/uploads/2021/05/inossi-got-you.mp3',
                            title: 'Got You',
                            artist: 'Noor',
                            artwork: 'https://picsum.photos/id/103/200/300',
                            album: 'INOSSI',
                        }

                    }
                },
            }

        }

    }


});
jest.mock('react-i18next', () => {
    return {
        useTranslation: () => {
            return {
                t: jest.fn(),
            };
        },
    };
});

jest.mock('react-native-iphone-x-helper', () => ({
    getStatusBarHeight: jest.fn(),
    getBottomSpace: jest.fn(),
}));
jest.mock('react-redux', () => ({
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
}));

// jest.mock('react-native-modal', () => 'react-native-modal');
jest.mock('react-native-modal', () => {
    return {
        __esModule: true,
        A: true,
        namedExport: jest.fn(),
        default: 'mockedDefaultExport',
    };
});
describe("group tesing goes here", () => {
    test('change text', () => {
        const { getByTestId } = render(<AppCreatePlaylistModal />);
        // fireEvent.press(getByTestId('changeTest'));
        // fireEvent(getByPlaceholderText('changeTest'), 'onChangeText', 'ab');
        fireEvent.changeText(getByTestId('changeTest'));
    });
    test('closing modal', () => {
        const { getByTestId } = render(<AppCreatePlaylistModal />);
        fireEvent.press(getByTestId('modalClose'));
    });
    test('closing modal', () => {
        const { getByTestId } = render(<AppCreatePlaylistModal />);
        fireEvent.press(getByTestId('savingNewPlayList'));
    });
})

