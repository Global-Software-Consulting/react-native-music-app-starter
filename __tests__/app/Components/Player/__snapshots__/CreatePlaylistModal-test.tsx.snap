// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group tesing goes here renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#03dac4",
      "justifyContent": "center",
      "paddingLeft": 24,
      "paddingRight": 24,
    }
  }
>
  <react-native-modal>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#03dac4",
            "borderRadius": 10,
            "height": "50%",
            "width": "90%",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#03dac4",
            "borderBottomColor": "lightgray",
            "elevation": 5,
            "flexDirection": "row",
            "height": undefined,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "padding": 8,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <mockedDefaultExport
              color="#6200ee"
              name="close"
              size={30}
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "fontSize": undefined,
                  "fontWeight": "bold",
                  "marginBottom": 12,
                  "paddingLeft": "5%",
                },
              ]
            }
          >
            Create Playlist
          </Text>
          <View
            style={
              Object {
                "width": "20%",
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#6200ee",
                "borderRadius": 5,
                "borderStartColor": "red",
                "flexDirection": "row",
                "height": undefined,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 5,
                "width": undefined,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "#03dac4",
                    "fontWeight": "bold",
                    "height": undefined,
                    "paddingTop": "5%",
                  },
                ]
              }
            >
              Save
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginRight": 25,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "fontSize": undefined,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "justifyContent": "space-between",
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          Playlist:
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="default"
          maxLength={10}
          onChangeText={[Function]}
          placeholder="Type here.."
          rejectResponderTermination={true}
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#6200ee",
              "color": "#6200ee",
              "height": 40,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </react-native-modal>
</View>
`;
